### ISSUE TITLE
Customs session names are broken

### ISSUE THREAD

**Issue by itkfm** [yiisoft/session#45]
### What steps will reproduce the problem?

```php
$v = new \Yiisoft\Session\Session([
    'name' => 'customname',
]);
```

### What is the expected result?

`Yiisoft\Session\Session::getName` would return `$this->options['name]` if available (or first check whether there’s an active session) before resorting to `session_name()`.

### What do you get instead?

a fresh session and new session cookie each response

### Additional info

`session_name()` always returns `PHPSESSID` when there’s been no session started yet.

| Q                | A
| ---------------- | ---
| Version          | 1.1.0
| PHP version      | 7.4
| Operating system | Debian GNU/Linux 11

**Comment by itkfm** [comment]
Why is this a problem?

The first thing `SessionMiddleware::process` does, is to call `::getSessionIdFromRequest(…)` which then results in a call to `Yiisoft\Session\Session::getName` which will always return `"PHPSESSID"` at this time

**Comment by vjik** [comment]
Fixed by #46
